Src/IronPython/Runtime/Operations/PythonTypeOps.cs

```diff
@@ -345,13 +345,35 @@ internal static class PythonTypeOps {
             return group;
         }
 
-        private static BuiltinFunction GetConstructor(Type t, bool privateBinding) {
+        private static BuiltinFunction GetConstructorFunction(Type t, bool privateBinding) {
             BuiltinFunction ctorFunc = InstanceOps.NonDefaultNewInst;
-            MethodBase[] ctors = CompilerHelpers.GetConstructors(t, privateBinding, true);
+            MethodBase[] ctors = PythonTypeOps.GetConstructors(t, privateBinding, true);
 
             return GetConstructor(t, ctorFunc, ctors);
         }
 
+        internal static MethodBase[] GetConstructors(Type t, bool privateBinding, bool includeProtected = false) {
+            MethodBase[] ctors = CompilerHelpers.GetConstructors(t, privateBinding, includeProtected);
+            if (t.IsEnum()) {
+                var enumCtor = typeof(PythonTypeOps).GetDeclaredMethods(nameof(CreateEnum)).Single().MakeGenericMethod(t);
+                ctors = ctors.Concat(new[] { enumCtor }).ToArray();
+            }
+            return ctors;
+        }
+
+        // support for EnumType(number)
+        private static T CreateEnum<T>(object value) {
+            try {
+                return (T)Enum.ToObject(typeof(T), value);
+            } catch (ArgumentException) {
+                throw PythonOps.ValueError(
```

Comment:
Please use string interpolation here:

```csharp
throw PythonOps.VarlueError("{value} is not a valid {typeof(T)}")
```

Reply:
You would get `null is not a valid ...` instead of `None is not valid ...`, are you sure?

Reply:
I was just using an example. Also, it may be better to check for null before the ToObject call. Have you tried without the try/catch and see what error is raised and shown without creating your own throw?

Reply:
for `null` it will throw an `ArgumentNullException` which would be caught by `catch (ArgumentException)`
for the exception message, I am using the one from `enum34`, and the one from .NET contains the words `null` and `Int32`...

Reply:
Yes, I understand, but exceptions are slow. If you check for null at the beginning and then throw your own ValueError for that, it will be faster in that case.

Reply:
what are the chances of someone actually doing `DaysInt(None)`?

Reply:
You'd be surprised!

Reply:
done

Reply:
anything else other than waiting for CI to pass?